### DISCUSSION TITLE
How to search in column that was added using addColumn

### DISCUSSION THREAD

**Discussion by cooperinho** [yajra/laravel-datatables#2758] [Q&A]
I have a few columns where I need to combine data of some columns into one and style them and add some text.

This is an added row for example:

```
    public function dataTable($query)
    {
        return datatables()
            ->eloquent($query)
            ->addColumn('status', function ($data) {                       
                       if ($data->status== '0') return "Not confirmed";
                       else return "Confirmed";

        // ...
  

    protected function getColumns()
    {
        return [           
            Column::computed('status')
            // ...
```

How to search in that row by the text? ("Confirmed" or "Not confirmed")?

**Comment by yajra** [comment]
If you want it to be searchable when using a `query`, you need to add the column in SQL level. This would also allow you to sort on that column.

Note that the snippet below is not tested and just an idea.

```php
->addSelect(DB::raw("CASE WHEN status = 0 THEN 'Not Confirmed' ELSE 'Confirmed' as status")

->filterColumn('status', function($q, $k) {
  // do counter part sql for search
})
```

See [docs](https://yajrabox.com/docs/laravel-datatables/master/filter-column)
